File: Algorithms/C/1290-Convert_Binary_Number_in_a_Linked_List_to_Integer.c
// Runtime: 0 ms, faster than 100.00% of C online submissions for Convert Binary Number in a Linked List to Integer.
// Memory Usage: 6.9 MB, less than 100.00% of C online submissions for Convert Binary Number in a Linked List to Integer.
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


int getDecimalValue(struct ListNode* head){
    int result = 0;
    while (head != NULL) {
        result = result * 2 + head->val;
        head = head->next;
    }
    return result;
}

File: Algorithms/Python3.x/1290-Convert_Binary_Number_in_a_Linked_List_to_Integer.py
# Runtime: 20 ms, faster than 100.00% of Python3 online submissions for Convert Binary Number in a Linked List to Integer.
# Memory Usage: 12.7 MB, less than 100.00% of Python3 online submissions for Convert Binary Number in a Linked List to Integer.
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        result = 0
        while head is not None:
            result = result * 2 + head.val
            head = head.next
        return result
